### PR TITLE
New program to process `iftop` log output

### PR DESCRIPTION
**Pull request by pgarg66**
#### Problem
Need a utility to capture inbound and outbound traffic for each connection on network nodes. This will be useful to analyze packet drops in Solana cluster.

#### Summary of Changes
This PR adds a new program that consumes `iftop` log file, and outputs the latest traffic numbers for each connection on the nodes. The output file can be used to match how much data was lost after it was transmitted on individual connections.

For example, the following script downloads the processed logs from each node in the testnet, and stores the log in <ip-address>-iftop.log file.

```
source net/config/config
for ip in "${validatorIpList[@]}"; do :
./net/ssh.sh solana@$ip 'PATH=$PATH:~/.cargo/bin/ ~/solana/scripts/iftop-postprocess.sh ~/solana/iftop.log /tmp/temp.log' $ip > $ip-iftop.log
done
```

Fixes #

### CODE REVIEW

File: scripts/iftop.sh [hunk 1]
```diff
@@ -8,9 +8,11 @@ set -e
 
 cd "$(dirname "$0")"
 
-sudo iftop -i "$(ifconfig | grep mtu | grep -iv loopback | grep -i running | awk 'BEGIN { FS = ":" } ; {print $1}')" -nNbBP  -t \
-	| awk '{ if ($3 ~ "=>") { print $2, $7 } else if ($2 ~ "<=") { print $1, $6 }} ' \
-	| awk 'NR%2{printf "%s ",$0;next;}1' \
-	| awk '{ print $1, "<=>", $3, ":",  $2, $4 }'
+Sudo=
+if sudo true; then
+Sudo="sudo -n"
```
Comment (mvines) [inline scripts/iftop.sh]: nit: just `sudo="sudo -n"`, no S

File: scripts/iftop-postprocess.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  | awk 'NR%2{printf "%s ",$0;next;}1' \
+  | awk '{ print "{ \"a\": \""$1"\", " "\"b\": \""$3"\", \"a_to_b\": \""$2"\", \"b_to_a\": \""$4"\"}," }' > "$2"
+
+if [ "$#" -ne 3 ]; then
+  solana-network-tool -i "$2"
+else
+  solana-network-tool -i "$2" map-IP --priv "$(hostname -i)" --pub "$3"
+fi
+
+exit 1
```
Comment (mvines) [inline scripts/iftop-postprocess.sh]: why `exit 1`?

File: scripts/iftop-postprocess.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+#
+# Reports network bandwidth usage
+#
```
Comment (mvines) [inline scripts/iftop-postprocess.sh]: can you please document the expected command-line arguments

File: scripts/iftop-postprocess.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+set -e
+
+cd "$(dirname "$0")"
+
+awk '{ if ($3 ~ "=>") { print $2, $7 } else if ($2 ~ "<=") { print $1, $6 }} ' < "$1" \
```
Comment (mvines) [inline scripts/iftop-postprocess.sh]: roll this awk into `solana-network-tool`?

File: scripts/cargo-install-all.sh [hunk 5]
```diff
@@ -52,6 +52,7 @@ BINS=(
   solana-install-init
   solana-keygen
   solana-ledger-tool
+  solana-network-tool
```
Comment (mvines) [inline scripts/cargo-install-all.sh]: maybe call this `iftop-post-processer`?  That's what it is right?

File: network-tool/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,22 @@
+[package]
+authors = ["Solana Maintainers <maintainers@solana.com>"]
+edition = "2018"
+name = "solana-network-tool"
+description = "The solana cluster network analysis tool"
+version = "0.1.0"
+repository = "https://github.com/solana-labs/solana"
+license = "Apache-2.0"
+homepage = "https://solana.com/"
```
Comment (mvines) [inline network-tool/Cargo.toml]: let's mark this package as `publish = false`

File: scripts/iftop-postprocess.sh (hunk 4 above)
Comment (pgarg66) [reply scripts/iftop-postprocess.sh]: I was hoping to roll in more functionality into the tool. The arguments can decide what to do.

File: scripts/cargo-install-all.sh (hunk 5 above)
Comment (pgarg66) [reply scripts/cargo-install-all.sh]: I was hoping to roll in more functionality into the tool. The arguments can decide what to do.

File: scripts/iftop-postprocess.sh (hunk 4 above)
Comment (pgarg66) [reply scripts/iftop-postprocess.sh]: I'll look into it. Ok to do it in a follow up PR?

Comment (mvines) [reply scripts/iftop-postprocess.sh]: follow-up is fine with me

File: scripts/cargo-install-all.sh (hunk 5 above)
Comment (pgarg66) [reply scripts/cargo-install-all.sh]: renamed to `log-analyzer`

**Pull request merged**